### DISCUSSION TITLE
Add commands to end of PS1 prompt?

### DISCUSSION THREAD

**Discussion by ProgRick** [JanDeDobbeleer/oh-my-posh#3294] [Ideas]
I am using oh-my-posh in git-bash on my Windows laptop. Typically I have several git-bash sessions opened at a time and I would like to have the commands I type in any tab be part of the history anywhere else. Normally this can be done by adding the following commands to the end of the prompt (PS1) : history -a; history -c; history -r

Is there anyway to be able to add these commands to the prompt that oh-my-posh creates? It would be great if there were a configuration variable that would append commands like that.

**Comment by JanDeDobbeleer** [comment]
@ProgRick you can override `set_poshcontext` to execute logic during every prompt invocation (which should result in the same):

```bash
function set_poshcontext() {
  history -a
  history -c
  history -r
}
```

**Reply by ProgRick** [reply]
Ok. I guess you might need to give me some more help with this. Where do I put that function to have it take affect? Do I need to download the source  and rebuild the exe? Or is there a file I can add that function into?

**Reply by JanDeDobbeleer** [reply]
@ProgRick you can add it to your `bashrc` after initializing oh-my-posh.

**Reply by ProgRick** [reply]
Thanks that works. I was adding it before the init.
